docs/cpp-runnable.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Generating Executables
+id: cpp-runnable
+---
+
+The `fuse` CLI provides the `run` subcommand for generating executables that
+can be used to test Fuse code without going throught a synthesis workflow.
+
+To build an executable, first run the following command from the project root:
+
+```
+sbt assembly
+```
+
+This will rebuild the compiler and download any headers required for building
+the executable.
+
+## Example
+
+As an example, we'll be using the following fuse code:
+
+```C
+decl a: float[2][2];
+decl b: float[2][2];
+decl c: float[2][2];
+
+for (let i = 0..2) {
+  for (let j = 0..2) {
+    c[i][j] := a[i][j] + b[i][j];
+  }
+}
+```
+
+Add this code to a file titled `matadd.fuse`.
+
+Next, run the command:
+
+```
+fuse run matadd.fuse -o matadd.cpp
+```
+
+the command will something similar to the following:
+
+```sh
+g++ --std=c++11 -Wall -Isrc/main/resources/headers matadd.cpp -o matadd.cpp.o
+```
```

Comment:
This wording seems to suggest that `fuse run` *just* executes `g++`. But that's not quite true: it first compiles the source to C++ (including the wrapper) and then compiles it the rest of the way. I'd just say that, describing all the things the command does—and maybe even omit this command, the precise details of which are maybe not terribly relevant.

Reply:
#114 fixed this.